cpp/src/io/json/json_gpu.cu

```diff
@@ -791,31 +792,31 @@ std::vector<cudf::io::column_type_histogram> detect_data_types(
   CUDA_TRY(
     cudaOccupancyMaxPotentialBlockSize(&min_grid_size, &block_size, detect_data_types_kernel));
 
-  rmm::device_vector<cudf::io::column_type_histogram> d_column_infos(
-    num_columns, cudf::io::column_type_histogram{});
-
-  if (do_set_null_count) {
-    // Set the null count to the row count (all fields assumes to be null).
-    thrust::for_each(
-      rmm::exec_policy(stream),
-      d_column_infos.begin(),
-      d_column_infos.end(),
-      [num_records = row_offsets.size()] __device__(auto &info) { info.null_count = num_records; });
-  }
+  auto d_column_infos = [&]() {
+    if (do_set_null_count) {
+      rmm::device_uvector<cudf::io::column_type_histogram> d_column_infos(num_columns, stream);
+      // Set the null count to the row count (all fields assumes to be null).
+      thrust::generate(
```

Comment:
TIL. Substituting `for_each()` with `generate()`. I've to keep this in mind for my own use.